### PR TITLE
Remove string parsing; add AstroAngles.jl

### PR DESCRIPTION
**Pull request by mileslucas**
Removes the string parsing code that was ported to AstroAngles.jl and adds it as a dependency. Examples have been added to the docs. This will be breaking change.

### CODE REVIEW

**Comment by giordano** [comment]
I think we should do a release before this, last stable version is from 2019 and there have been already some changes since then: https://github.com/JuliaAstro/SkyCoords.jl/compare/v0.3.0...5ec8a65

File: docs/Project.toml [hunk 1]
```diff
@@ -1,4 +1,5 @@
 [deps]
+AstroAngles = "5c4adb95-c1fc-4c53-b4ea-2a94080c53d2"
 Documenter = "e30172f5-a6a5-5a46-863b-614d45cd2de4"
 Unitful = "1986cc42-f94f-5a68-af5c-568840ba703d"
```
Comment (giordano) [inline docs/Project.toml:5]: Raise compat of Documenter below

**Comment by mileslucas** [comment]
Okay, that sounds good, I'll release a `v0.6.1` off the current master branch

**Comment by mileslucas** [comment]
Okay, AstroAngles has julia compat of `^1.3`, you think it's fine to go to that here?

**Comment by giordano** [comment]
Yes, and I'd also go to v1.0 at this point, dealing with 0.x is kinda painful

**Comment by giordano** [comment]
Can we say this fixes #6 (even though some parsing was already present here)?

**Comment by mileslucas** [comment]
yeah, that seems appropriate. The only other thing I could think to add is something that automatically converts `AbstractQuantity` to radians to support `Unitful.°` inputs, but that's pretty trivial/not necessary to write in until someone asks for it.

**Pull request merged**